ntoskrnl/ps/quota.c

```diff
@@ -28,66 +28,62 @@ static KSPIN_LOCK PspQuotaLock;
 
 /*
  * Private helper to charge the specified process quota.
- * ReturnsSTATUS_QUOTA_EXCEEDED on quota limit check failure.
- * Updates QuotaPeak as needed for specified PoolIndex.
- * TODO: Research and possibly add (the undocumented) enum type PS_QUOTA_TYPE
- *       to replace UCHAR for 'PoolIndex'.
+ * Returns STATUS_QUOTA_EXCEEDED on quota limit check failure.
+ * Updates QuotaPeak as needed for specified quota type in PS_QUOTA_TYPE enum.
  * Notes: Conceptually translation unit local/private.
  */
 NTSTATUS
 NTAPI
 PspChargeProcessQuotaSpecifiedPool(IN PEPROCESS Process,
-                                   IN UCHAR     PoolIndex,
+                                   IN PS_QUOTA_TYPE QuotaType,
                                    IN SIZE_T    Amount)
 {
     ASSERT(Process);
     ASSERT(Process != PsInitialSystemProcess);
-    ASSERT(PoolIndex <= 2);
+    ASSERT(QuotaType < PsQuotaTypes);
```

Comment:
NT 5.1/5.2:
```suggestion
    ASSERT(QuotaType < 3); // See EPROCESS.
```

NT 6.0:
postponed.

Reply:
lol

With your comment you're telling me to go back using magic values, that's a big no.

Reply:
Not really: I am merely suggesting to be consistent with geoffchappell documentation (here and elsewhere), unless the latter is proven wrong...